src/main/java/pl/kurs/watercontainers/models/WaterContainer.java

```diff
@@ -0,0 +1,65 @@
+package pl.kurs.watercontainers.models;
+
+import java.util.Objects;
+
+public class WaterContainer {
+
+
+    private String name;
+    private double maxCapacity;
+    private double waterLevel;
+
+    public WaterContainer(String name, double maxCapacity, double waterLevel) {
```

Comment:
set this constructor to private and add static factory method with validation like "waterLevel cant be below 0" etc